### PR TITLE
Link release notes in the Documention menu

### PR DESCRIPTION
**Pull request by sleto-it**
### Description

Add the recently created release notes page in the `Documentation` menu. Reference: https://github.com/valkey-io/valkey-doc/pull/171
 
### Issues Resolved

No GH issue open.

### Check List
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: content/docs/index.md [hunk 1]
```diff
@@ -12,3 +12,4 @@ page_template = "docs-page.html"
     * [ACL](/topics/acl/)
 * Valkey Manual
   * [Keyspace Notifications](/topics/keyspace/)
+* [Release Notes](topics/release-notes/)
```
Comment (stockholmux) [inline content/docs/index.md:15]: humm. I don't mind the idea of linking the release notes but `/topics/release-notes/` isn't a documentation topic in valkey-doc

**Review by stockholmux — CHANGES_REQUESTED** [review]
we need a way to link the release notes, not just a link

File: content/docs/index.md (hunk 1 above)
Comment (sleto-it) [reply content/docs/index.md:15]: Hi @stockholmux ! Thanks for reviewing this!  When I created this PR it was in combination with https://github.com/valkey-io/valkey-doc/pull/171 . If https://github.com/valkey-io/valkey-doc/pull/171 is not going to be included, then we can close this one. Hope this helps

**Comment by stockholmux** [comment]
I'm going to close this one in lieu of work on #145

**Pull request closed without merging**